server/src/main/java/org/elasticsearch/cluster/ClusterChangedEvent.java

```diff
@@ -163,6 +163,25 @@ public Set<String> changedCustomProjectMetadataSet() {
         return result;
     }
 
+    /**
+     * Returns a set of custom meta data types when any custom metadata for the project metadata has changed
+     * between the previous cluster state and the new cluster state. custom metadata types are
+     * returned iff they have been added, updated or removed between the previous and the current state
+     */
+    public Set<String> changedCustomProjectMetadataSet(ProjectId projectId) {
```

Comment:
Since you're adding a new method anyway, could you incorporate the TODO above in this method:
```
        // TODO: none of the usages of these `changedCustom` methods actually need the full list; they just want to know if a specific entry
        // changed.
```
In other words, can you change the signature of the method to:
```java
public boolean projectMetadataCustomChanged(String custom, ProjectId projectId)
```